zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -549,6 +549,36 @@ pub fn standard_coinbase_outputs(
         .collect()
 }
 
+/// Returns a fake coinbase transaction that can be used during transaction selection.
+///
+/// This avoids a data dependency loop involving the selected transactions, the miner fee,
+/// and the coinbase transaction.
+///
+/// This transaction's serialized size and sigops must be at least as large as the real coinbase
+/// transaction with the correct height and fee.
+fn fake_coinbase_transaction(
```

Comment:
>     * set the returned value as a field on `GetBlockTemplateRpcImpl` so it's only computed once

The size of the coinbase outputs depends on the block height, so this function is using the wrong height, and we can't store the result. (We can only store the result if all future blocks don't add funding streams, but there is a rule in ZIP-1017 that allows a large grant to be paid out via an extra funding stream.)

I'll fix this in my next PR.

>     * rename this to something like `coinbase_transaction_allocations` and only return the `sigops`/`data.len`

Thanks, I'll do this in my next PR.

Reply:
> > ```
> > * set the returned value as a field on `GetBlockTemplateRpcImpl` so it's only computed once
> > ```
> 
> The size of the coinbase outputs depends on the block height, so this function is using the wrong height, and we can't store the result. (We can only store the result if all future blocks don't add funding streams, but there is a rule in ZIP-1017 that allows a large grant to be paid out via an extra funding stream.)

I went and fixed this bug now, because it removes the dependency on PR #5787 getting reviewed and merged. See 018bf32

I also used a 1 zat fee, just in case someone gets clever and starts skipping zero outputs. See b2e8c0e

Reply:
> * rename this to something like `coinbase_transaction_allocations` and only return the `sigops`/`data.len`

I decided not to do this, because the fake coinbase transaction is typed, but the fields are `u64` and `usize`. So it would be easy to get them mixed up with the real fields for the template or other transactions.